youtube_transcript_api/formatters.py

```diff
@@ -134,13 +134,66 @@ def format_transcripts(self, transcripts, **kwargs):
         """
         return '\n\n\n'.join([self.format_transcript(transcript, **kwargs) for transcript in transcripts])
 
+class SRTFormatter(Formatter):
+    def _seconds_to_timestamp(self, time):
+        """Helper that converts `time` into a transcript cue timestamp for SRT.
+
+        :param time: a float representing time in seconds.
+        :type time: float
+        :return: a string formatted as a cue timestamp, 'HH:MM:SS,MS'
+        :rtype str
+        :example:
+        >>> self._seconds_to_timestamp(6.93)
+        '00:00:06,930'
+        """
+        time = float(time)
+        hours, remainder = divmod(time, 3600)
+        mins, secs = divmod(remainder, 60)
+        ms = int(round((time - int(time))*1000, 2))
+        return "{:02.0f}:{:02.0f}:{:02.0f},{:03d}".format(hours, mins, secs, ms)
+
+    def format_transcript(self, transcript, **kwargs):
+        """Converts a transcript into SRT formatting.
+
+        :param transcript:
+        :reference: https://www.3playmedia.com/blog/create-srt-file/
+        """
+        lines = []
+        for i, line in enumerate(transcript):
+            if i < len(transcript) - 1:
+                # Looks ahead, use next start time since duration value
+                # would create an overlap between start times.
+                time_text = "{} --> {}".format(
+                    self._seconds_to_timestamp(line['start']),
+                    self._seconds_to_timestamp(transcript[i + 1]['start'])
```

Comment:
I didn't realize this when approving the initial PR for WebVTT, but it doesn't seem to make sense to me to always use the end time of the next start time, even if `transcript[i + 1]['start'] > line['start'] + line['duration']`. I guess this whole if-else could be replaced by
```python
end = line['start'] + line['duration']
time_text = "{} --> {}".format(
  self._seconds_to_timestamp(line['start']),
  self._seconds_to_timestamp(
    transcript[i + 1]['start'] if i < len(transcript) - 1 and transcript[i + 1]['start'] < end else end
  )
)
```

Reply:
I also changed this section of code with your suggested code.